internal/sys/fs.go

```diff
@@ -227,7 +245,11 @@ type FileEntry struct {
 	// Name is the name of the directory up to its pre-open, or the pre-open
 	// name itself when IsPreopen.
 	//
-	// Note: This can drift on rename.
+	// # Notes
+	//
+	//   - This can drift on rename.
+	//   - This relates to the guest path, which is not the real file path
```

Comment:
I made notes here because the result here (used by wasi) is not the file's real path, rather it is about the what the guest sees as the path.